application/controllers/TemplateController.php

```diff
@@ -20,7 +22,12 @@ public function indexAction()
     {
         $this->createTabs()->activate('preview');
 
-        $template = Template::fromDb($this->params->getRequired('id'));
+        /** @var $template Model\Template */
+        $template = Model\Template::on($this->getDb())
+            ->filter(Filter::equal('id', $this->params->getRequired('id')))
+            ->first();
+
+        $template = Template::fromModel($template);
```

Comment:
Please do this after the below `if` block which checks if the template is `null`.